### PR TITLE
Issue #3763: List Min/Max Problems

### PR DESCRIPTION
**Pull request by hawkfish**
* Fix list comparison typo
* Always use default incremental selection vector when Orrifying flat vectors.

### CODE REVIEW

File: src/common/types/vector.cpp [hunk 1]
```diff
@@ -764,7 +764,7 @@ void Vector::Orrify(idx_t count, VectorData &data) {
 		break;
 	default:
 		Normalify(count);
-		data.sel = FlatVector::IncrementalSelectionVector(count, data.owned_sel);
+		data.sel = FlatVector::IncrementalSelectionVector();
```
Comment (Mytherin) [inline src/common/types/vector.cpp:767]: Should we perhaps modify `FlatVector:: IncrementalSelectionVector` to ignore the count and always return `FlatVector::IncrementalSelectionVector`?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! LGTM, one comment:

File: src/common/types/vector.cpp (hunk 1 above)
Comment (hawkfish) [reply src/common/types/vector.cpp:767]: I think this gets back to the whole issue of `const`ness. From that perspective, `FlatVector:: IncrementalSelectionVector()` should return a `const SelectionVector*`. This then implies that `VectorData.sel` should also be a `const SelectionVector*`. It also suggests that `FlatVector:: IncrementalSelectionVector(count,...)` should _always_ populate the passed `SelectionVector` and return a writable `SelectionVector*`.

This is a bigger change that I'm happy to do, but its a lot more work than fixing the failures. Shall we break it up into two parts?

**Review by taniabogatsch — COMMENTED** [review]
Looks good to me. :)

**Pull request merged**